// kts-invoice-engine/InvoiceEngine/HangXin/issueEInvoice/INonVehicle.ts
/** 开具专普电票 */
export default interface IData {
    /**
     * 分机号
     */
    fjh: string;
    /**
     * 销方税号
     */
    taxno: string;

    FPKJXX: IKjxx;
}
interface IKjxx {
    FPQQLSH: string;
    DDH: string;
    /**
     * 发票种类 
     * 0:专票,2:普票 51:电子票
     */
    FPZL_DM: string;
    /**
     * 开票类型
     * 0-蓝字发票;1-红字发票
     */
    KPLX: string;
    XHF_NSRSBH: string;
    XHFMC: string;
    XHF_DZ: string;
    XHF_YHZH: string;
    /**
     * 购方税号
     * 专票不能为空
     */
    GHF_NSRSBH: string;
    GHFMC: string;
    GHF_DZ: string;
    GHF_YHZH: string;
    GHF_SJ: string;
    GHF_EMAIL: string;
    KPY: string;
    SKY: string;
    FHR: string;
    YFP_DM: string;
    YFP_HM: string;
    HZTZD: string;
    /**
     * 清单标志
     * 0-无清单;1-有清单。电票时 为空，纸票必须传值
     */
    XHQD: string;
    QDMC: string;
    KPDMC: string;
    BMB_BBH: string;
    BZ: string;
    /**
     * 营业税标志
     * 
     */
    YYSBZ: string;
    FPZDYLX: string;
    BYZD1: string;
    BYZD2: string;
    TSFS: string;
    SPBMJCBZ: string;
    FPKJXX_XMXXS?: IXmxxs[];
}
interface IXmxxs {
    FPHXZ: string;
    XMMC: string;
    GGXH: string;
    XMDW: string;
    XMSL: string;
    XMDJ: string;
    XMJE: string;
    SL: string;
    SE: string;
    HSBZ: string;
    SPBM: string;
    ZXBM: string;
    YHZCBS: string;
    YHZCMC: string;
    LSLBS: string;
    ZZSTSGL: string;
    KCE: string;
    SM: string;
    BYZD1: string;
    BYZD2: string;
}
