// kts-invoice-engine/InvoiceEngine/HangXin/issueEInvoice/IVehicle.ts
/** 开具机动车票 不用*/
export default interface IData {
    /**
     * 分机号
     * 2 位分机号+2 位终端号
     */
    fjh: string;
    /**
     * 销方税号
     * 真实税号是 15/18 位
     */
    taxno: string;

    FPKJXX: IKjxx;
}
interface IKjxx {
    /**
     * 发票种类
     * 12：机动车
     */
    InfoKind: string;
    BillNumber: string; 
    InfoClientName: string; 
    InfoClientTaxCode: string; 
    InfoClientAddressPhone: string; 
    IDCard: string;
    VehicleKind: string;
    BrandModel: string;
    OriginPlace: string;
    QualityCertificate: string;
    ImpCertificateNo: string; CommInspectionNo: string;
    EngineNo: string;
    VehicleNo: string;
    ManufacturerName: string; AmountTaxTotal: string;
    SellerPhone: string;
    SellerAccount: string;
    SellerAddress: string;
    SellerBank: string;
    InfoTaxRate: string;
    Tonnage: string;
    PeopleNo: string;
    BlueInfoTypeCode: string;
    BlueInfoNumber: string;
    GoodsNoVer: string;
    GoodsTaxNo: string;
    /**
     * 优惠政策标识
     * 0:不享受，1:享受
     */
    TaxPre: string;
    /**
     * 优惠政策名称
     * 免税、出口退税、不征收、普 通零税率、空
     */
    TaxPreCon: string;
    ZeroTax: string;
    CropGoodsNo: string;
    TaxDeduction: string;
    Drawer: string;
}
