// kts-invoice-engine/InvoiceEngine/InvoiceMap.ts
export interface InvoiceMap {
    invoiceCode?: string;//发票代码
    invoiceNo?: string;//发票号码
    invoiceType?: string;//发票类型
    /**
     * 清单模式(0~8 设定明细条数超过转清单，0 始终是清单模式)	
     */
    inventory?:string
    /**
     * 单据编号
     */
    billingData?:string;
    /**
     * 开票类型 0-正数发票;1-负数发票
     */
    typeCode?: string;
    /**
     * 机器编号
     */
    machineNo?: string;
    /**
     * 开票日期
     */
    issueDate?: string;
    taxControlCode?: string;//发票密文
    /**
     * 主要商品名称
     */
    mainGoodsName?:string;
    /**
     * 发票校验码
     */
    invoiceCheckCode?: string;
    graphCode?: string;//二维码
    invoiceSIA1?: string;//发票特殊标识区域 1
    invoiceSIA2?: string;//发票特殊标识区域 2
    /**
     * 收款人
     */
    payee?: string;
    /**
     * 复核
     */
    checker?: string;
    /**
     * 开票人
     */
    invoiceClerk?: string;
    /**
     * 价税合计
     */
    taxInclusiveTotalAmount?: string;
    /**
     * 合计金额
     */
    taxExclusiveTotalAmount?: string;
    /**
     * 合计税额 
     */
    taxTotalAmount?: string;
    /**
     * 备注
     */
    note?: string;
    goodsInfos?: IGoodsInfo[];//发票明细
    //Extention?: string;//发票数据扩展
    signature?: string;//签名值
    version?: string;//
    namespace?: string;//
    /**
     * 销售方纳税人识别号
     */
    sellerTaxID?: string;
    /**
     * 销售方名称
     */
    sellerName?: string;
    /**
     * 销售方地址、电话
     */
    sellerAddrTel?: string;
    /**
     * 销售方银行账号
     */
    sellerFinancialAccount?: string;
    /**
     * 购买方纳税人识别号
     */
    buyerTaxID?: string;
    /**
     * 购买方名称
     */
    buyerName?: string;
    /**
     * 购买方地址、电话
     */
    buyerAddrTel?: string;
    /**
     * 购买方银行账号
     */
    buyerFinancialAccount?: string;
    /**
     * 税率
     */
    comprehensiveTaxRate?:string;
}
export interface IGoodsInfo {
    /**
     * 项目名称
     */
    item?: string;
    /**
     * 单位
     */
    measurementDimension?: string;
    /**
     * 规格型号
     */
    specification?: string;
    /**
     * 数量
     */
    quantity?: string;
    /**
     * 单价
     */
    price?: string;
    /**
     * 金额
     */
    amount?: string;
    /**
     * 税率
     */
    taxScheme?: string;
    /**
     * 税额
     */
    taxAmount?: string;
    extention?: string;

    /**
     * 发票行性质：0－正常发票行；3－被折扣行；4－折扣行
     * 折扣行是前端计算的
     */
    lineDiscount?: string;
    /**
     * 商品税目
     */
    goodsTaxItem?: string;
    /**
     * 税务编码 19位
     */
    taxCode?: string;
    /**
     * 含税标志 (0-不含税单价 1-含税单价  2-不含税单价＋含税金额 3-含税单价＋含税金额)
     * 3:因为金额(amount)里面已经包含了税额，所以商品行的税额(taxAmount)应该为0，
     * 0:计算出商品行的税额
     */
    taxSign?: string;
    /**
     * 商品编码
     */
    goodsCode?: string;
    /**
     * 折扣金额
     */
    discountAmount?:string;

    /**
     * 免税类型    0正常税率 1出口免税和其他免税优惠政策 2不征增值税 3普通零税率
     */
    taxRateSign?:string;
    /**
     * 0税类型 (0 0%  1 免税  2 不征税)
     */
    freeType?:string;
    /**
     * 优惠政策标识
     * 比如:超税负3%即征即退
     */
    bonusInfo?:string;
}
